### ISSUE TITLE
Cannot get the app to run on the AVD

### ISSUE THREAD

**Issue by nushydude** [CodeLanka/ez-net-app#10]
Cannot get the app to run on the AVD. How to solve this? 

# Steps followed 

- Fork to my github account
- CD to my local root folder
`cd ~/Documents/Github`
- Clone the project using:
`git clone https://github.com/nushydude/ez-net-app.git`
- Go into the app and install the dependencies
`cd ez-net-app`
`npm install` 
- Open the project and run AVD 
Open Android Studio 2.3.3
Open an existing Androind Studio project from initial screen
Point to ~/Documents/GitHub/ez-net-app
Android Studio says "Frameworkd detected. Androind framework is detected in the project. Configure."
Click on Configure
Click OK (There are already checked checkboxed called `Android` and `Android -> 9 AndroidManifest.xml files`)
Start AVD (Nexus 5 API 23 running Android 6.0 x86_64)
- After the AVD boots up, run `npm start` to start the app.
Build seems to go fine as I get the following screen.
![Alt text](https://lh3.googleusercontent.com/-FFzDln-2gNo/WViNZa9llUI/AAAAAAABQNc/Jzn0XpdyiVwI1gy9pnoaRqQFdK7FK-MggCHMYCw/s0/mintty_2017-07-02_16-06-27.png "Optional title")

But nothing shows up on the AVD.

# Environment
Windows 10 x64
Android Studio 2.3.3
Node.js 8.1.2

**Comment by nushydude** [comment]
OK, it seems the command you have to run is `react-native run-android` not `npm start`.

But I ran into this error. 

![Alt text](https://lh3.googleusercontent.com/-e41EwtiDCIU/WViUEmxNnpI/AAAAAAABQNo/Bw9hrswJu_YKc_5YHU4VkuDbwnZ1LewFwCHMYCw/s0/2017-07-02_16-34-48.png "Optional title")

**Comment by nushydude** [comment]
So the problem seems to be caused by native-base package. If I remove the Drawer component and native-base components used in the App component, the problem goes away.

**Comment by agentmilindu** [comment]
Hi @nushydude, What is the `native-base` version you have?

**Comment by nushydude** [comment]
@agentmilindu One in the package.json. v2.0.12.

**Comment by agentmilindu** [comment]
@nushydude That's weird, I have no clue why this is.

I worried if something is wrong with the code, but I got Travis CI building the bundle and APK ( https://travis-ci.org/CodeLanka/ez-net-app ) without errors, using both Node 6.9 and 7.1. And if you install the latest release ( https://github.com/CodeLanka/ez-net-app/releases/latest ) on your phone, which was built and pushed by Travis CI, you can see it is running without any error.


I see lot of people has came across  this `cannot read property 'symbol(symbol.iterator)' of null` error, but not related to `native-base` nor `react-native`. 

What shall we do? Can it be your Node version or react-native cli version. Shall we try updating them and see? Honestly, I don't have an idea about this issue.

**Comment by agentmilindu** [comment]
Since we have a `yarn.lock` file, if you use Yarn to install the dependencies instead of npm, version specific errors should disappear. Shall we try `yarn`?

**Comment by nushydude** [comment]
@agentmilindu I have a feeling that it has to do with the React-Native version I have installed globally. If I create a new app using `react-native init <app_name>` and install the dependencies by name (i.e. the latest versions will get installed), then I don't have any issue at all. It would work perfectly.

Do you run 'npm start' to build and launch the app? Does that start the app in your AVD? Or are you using a physical device to test the app? 

It was also complaining that the `package.json.lock` was not in the repo. Perhaps you need to include yours and this might fix the issue automatically? I honestly don't know what it does. But if the name is anything to go by, then is is 'locking' something so things do not get broken :) I am pretty new to React-"Native" so I don't know how people setup projects. I was just following a course on Udemy and I only know that method. 

The last time I installed `yarn`, it broke `react-native run-android` command. It required me to install `watcher` module through `yarn` every time I created a new app. I can give it another shot.

**Comment by nushydude** [comment]
@agentmilindu Can you share your local package-lock.json file with me?

**Comment by agentmilindu** [comment]
Hi @nushydude , I added a `package-lock.json`, Travis Ci built it without trouble, can you try now?

**Comment by nushydude** [comment]
@agentmilindu No improvement. :(

**Comment by agentmilindu** [comment]
Hi, Let's give this one last try. Let's go with Yarn.
First get a fresh clone, then do a `yarn`, then `yarn start` ( or `react-native start` ). Then in another terminal run `react-native run-android`, make sure you have an Android emulator running. 

You might need to install Watchman ( https://facebook.github.io/watchman/docs/install.html#installing-from-source )

**Comment by nushydude** [comment]
`yarn` messes things up for me. At least it did on my desktop. 

On the laptop, I have `npm` on last try. And I could get it to work. 

I don't know what fixed the problem. 

There were two things I did. 

1. I merged the latest code from the `master` branch to my `forked` branch. 
2. I also ran `npm start` first and then `react-native run-android`. Previously, I ran either. Usually the latter command is sufficient.

Finally I can do something.
